packages/type-utils/src/TypeOrValueSpecifier.ts

```diff
@@ -146,6 +146,19 @@ function typeDeclaredInFile(
   );
 }
 
+function typeDeclaredInLib(
+  declarationFiles: ts.SourceFile[],
+  program: ts.Program,
+): boolean {
+  // Intrinsic type (i.e. string, number, boolean, etc).
+  if (declarationFiles.length === 0) {
```

Comment:
Hmm, yeah, +1 to @marekdedic's request of testing. I'd also want to see what happens for error types. My intuition this should return `false`, perhaps?

Reply:
Finally came back to finish this. Added tests and added a check for error types.